src/js/module/K2previousissues/PreviousIssuesButtons.js

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import _ from 'underscore';
+import moment from 'moment';
+import BtnGroup from '../../component/BtnGroup';
+import * as API from '../../lib/api';
+
+const previousIssueButtons = [
+    {
+        title: 'Previous Version (-1M)',
+        ariaLabel: 'one month ago',
+        monthsAgo: 1,
+        type: 'monthly',
+    },
+    {
+        title: 'Previous Version (-3M)',
+        ariaLabel: 'three months ago',
+        monthsAgo: 3,
+        type: 'monthly',
+    },
+    {
+        title: 'Previous Version (-1Q)',
+        ariaLabel: 'one quarter ago',
+        quartersAgo: 1,
+        type: 'quarterly',
+    },
+    {
+        title: 'Previous Version (-1Y)',
+        ariaLabel: 'one year ago',
+        yearsAgo: 1,
+        type: 'all',
+    },
+];
+
+class PreviousIssuesButtons extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isOpen: false,
+            errorMessage: '',
+            showError: false,
+        };
+    }
+
+    showErrorMessage(message) {
+        this.setState({
+            errorMessage: message,
+            showError: true,
+        });
+
+        setTimeout(() => {
+            this.setState({
+                showError: false,
+            });
+
+            setTimeout(() => {
+                this.setState({
+                    errorMessage: '',
+                });
+            }, 300);
+        }, 5000);
+    }
+
+    openIssue(buttonConfig) {
+        API.getCurrentIssueDescription()
+            .then((issueTitleResponse) => {
+                const ghTitle = issueTitleResponse.data.title;
+
+                // Single regex to catch both Monthly and Quarterly formats
+                // The (\w+) between "Accounting" and "Close" captures either "Monthly" or "Quarterly"
+                const searchParts = ghTitle.match(/\[.*\] (Accounting (\w+) Close) ((?:\w{3} \d{4})|(?:Q\d \d{4})) (.*) \[.*\]/);
+
+                if (!searchParts) {
+                    console.debug('Found no search parts in issue title:', ghTitle);
+                    this.showErrorMessage('Unable to parse issue title format');
+                    return;
+                }
+
+                const closeType = searchParts[2]; // "Monthly" or "Quarterly"
+                const isQuarterly = closeType?.toLowerCase() === 'quarterly';
```

Comment:
Merp, my bad 🤷 Might have to go with the old school way

```suggestion
                const isQuarterly = (closeType || '').toLowerCase() === 'quarterly';
```